test/Image.js

```diff
@@ -0,0 +1,37 @@
+const Future = require("fibers/future");
+const compare = require("resemblejs").compare;
+
+module.exports.Image = class Image {
+  constructor(name, buffer) {
+    this.name = name;
+    this.buffer = buffer;
+  }
+
+  toString() {
+    return `Image<${this.name}>`;
+  }
+
+  serialize() {
+    return [this.name, this.buffer];
+  }
+
+  compare(other) {
+    if (this.name !== other.name) return false;
+    if (Buffer.compare(this.buffer, other.buffer) === 0) return true;
+
+    const threshold = 5;
+    const diff = Future.fromPromise(
+      new Promise((resolve, reject) => {
+        compare(this.buffer, other.buffer, (err, data) => {
+          if (err) reject(err);
+          else resolve(data);
+        });
+      })
+    ).wait();
+
+    const r = diff.isSameDimensions && diff.rawMisMatchPercentage < threshold;
+
+    if (!r) console.log(this.name, diff);
+    return r;
+  }
+};
```

Comment:
I'm also not sure what's the motivation behind this whole class - why is it expected that images might differ by any percentage if the algorithm is deterministic?

Reply:
1. in case we may switch from sharp to another engine.
2. it provides better output when fails

Anyway, this method improves QoL than comparing binary files.

Reply:
Regarding 1. I think it's a good thing that it fails, we need a human to have a look at the images in case, say, updating the version of a dependency somehow changes the image.

As for making it easier to understand how images differ, I totally agree that the current approach is just horrible. As an anecdote, what I've done for [`webapp-webpack-plugin`](https://github.com/brunocodutra/webapp-webpack-plugin) was to write the images generated to a tmp folder and [compare them against fixtures using `dir-compare`](https://github.com/brunocodutra/webapp-webpack-plugin/blob/master/test/util.js#L61), see for example [this test case](https://github.com/brunocodutra/webapp-webpack-plugin/blob/master/test/default.test.js). Whenever images differ, I can simply go look into the tmp folder.

In any case, we need a consistent way to assert on every test case - is there a reason why only two test cases make use of this class?